### DISCUSSION TITLE
Saving/ returning plots

### DISCUSSION THREAD

**Discussion by EleniSmyrniou** [cemsbv/pygef#322] [Q&A] [answered]
In plotting.py wouldn't it make more sense to return the fig and the axes? 
That way the user can make modifications in the plot. This is arleady being done in plot_merge but not in plot_bore and plot_cpt

**Comment by RDWimmers** [comment]
We do not return the `matplotlib.pyplot.Figure` object in the `plot_cpt` and `plot_bore` methods because you can provide an `Axes` based on a user defined `Figure`. Therefore, the need of returning it becomes unnecessary. A `Figure` object is only created within the function when the user does not provide an `Axes`. If we decide to return the `Figure` the return type of the function change based on the user input.

**Comment by tlukkezen** [comment, marked as answer]
@EleniSmyrniou although we only return the axes in the `plot_cpt` and `plot_bore` functions, the Figure that is created under the hood is still active in matplotlib and can still be accessed by the general API

```
axes = plot_cpt(...)
pyplot.savefig(...)
```

This will allow you to save your figure until you create another `Figure`.

Returning just the axes allows for more flexibility in adjusting/appending to existing axes, while keeping a consistent function signature and it still allows you to access the Figure, so it seems like the best solution in our view.

For maximum control you can create your own Subplot figure and provide the axes as arguments to the plotting functions.

One thing that we could do is also provide this signature philosophy to the `plot_merge` function for consistency.

**Reply by EleniSmyrniou** [reply]
Ok thanks!
